src/wallet/test/fuzz/spend.cpp

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <key_io.h>
+#include <outputtype.h>
+#include <script/script.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/util/setup_common.h>
+#include <validation.h>
+#include <wallet/coincontrol.h>
+#include <wallet/spend.h>
+#include <wallet/test/util.h>
+#include <wallet/wallet.h>
+
+namespace wallet {
+namespace {
+
+TestChain100Setup* g_setup;
+
+/**
+ * Singleton wallet class ensures that only one
+ * instance of CWallet is created and reused as required.
+ * This increases Fuzzing efficiency by reducing the expense
+ * of creating a new `Descriptor Wallet` each time and deletes
+ * the pointer safely using the destructor.
+ */
+class WalletSingleton
+{
+public:
+    static WalletSingleton& GetInstance()
+    {
+        static WalletSingleton instance;
+        return instance;
+    }
+
+    CWallet& GetWallet()

```

Comment:
```suggestion
    const CWallet& GetWallet()

```

It's otherwise hard to tell if the global wallet instance is mutated. Same for `GetCoins()`.

If the global instance is mutated, the target is likely to be non-deterministic which hurts fuzzer efficiency. The target should behave exactly the same when given the same input, but that might not be the case if global state is used and modified.

(The suggested change might not compile if `wallet` or `available_coins` are used as non-const/mutated)

Reply:
Yeah, this should make it more deterministic. As you said, it's causing some compilation errors with other code. I'll force-push once I've fixed them.

Reply:
The `CreateTransaction()` and `FundTransaction()` want the `CWallet` parameter to be non-const. I'm struggling to find a work around but I'm trying.

Just wanted to clear this, I think the `CWallet` instance inside these functions doesn't seem to be directly mutated or changed anytime. So, do we really want to make it to const. Am I understanding this correct?

Reply:
I've thought about it, and I think a global Singleton might not be the smartest way to conserve resources. The `wallet` object definitely mutates during execution. I think I've to remove the Singleton pattern, but doing so will make the file very slow, as creating the `Descriptor Wallet` is really time-consuming.